### ISSUE TITLE
Exception while changing settings

### ISSUE THREAD

**Issue by anjannath** [code-ready/tray-windows#61]
The tray gets into an exception while trying to configure some settings, see screenshot of the exception

<img src="https://user-images.githubusercontent.com/8885742/116215461-5389b000-a765-11eb-921d-d854086ddcc4.PNG" alt="tray-json-error" width="500"/>

Exception debug logs:
```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ArgumentOutOfRangeException: Value of '0' is not valid for 'Value'. 'Value' should be between 'Minimum' and 'Maximum'.
Parameter name: Value
   at System.Windows.Forms.NumericUpDown.set_Value(Decimal value)
   at CRCTray.CrcSettingsForm.loadConfigurationValues(ConfigResult cr) in C:\projects\tray-windows\Forms\CrcSettingsForm.cs:line 65
   at CRCTray.CrcSettingsForm.getConfigurationAndResetChanged() in C:\projects\tray-windows\Forms\CrcSettingsForm.cs:line 28
   at CRCTray.CrcSettingsForm.ApplyButton_Click(Object sender, EventArgs e) in C:\projects\tray-windows\Forms\CrcSettingsForm.cs:line 171
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4300.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
CRC-tray
    Assembly Version: 0.4.0.0
    Win32 Version: 0.4.0.0
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/CRC-tray.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4300.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4320.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Text.Json
    Assembly Version: 5.0.0.1
    Win32 Version: 5.0.220.61120
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Text.Json.DLL
----------------------------------------
System.Memory
    Assembly Version: 4.0.1.1
    Win32 Version: 4.6.28619.01
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Memory.DLL
----------------------------------------
System.Text.Encodings.Web
    Assembly Version: 5.0.0.0
    Win32 Version: 5.0.20.51904
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Text.Encodings.Web.DLL
----------------------------------------
Microsoft.Bcl.AsyncInterfaces
    Assembly Version: 5.0.0.0
    Win32 Version: 5.0.20.51904
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/Microsoft.Bcl.AsyncInterfaces.DLL
----------------------------------------
System.Threading.Tasks.Extensions
    Assembly Version: 4.2.0.1
    Win32 Version: 4.6.28619.01
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Threading.Tasks.Extensions.DLL
----------------------------------------
System.ValueTuple
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ValueTuple/v4.0_4.0.0.0__cc7b13ffcd2ddd51/System.ValueTuple.dll
----------------------------------------
System.Runtime.CompilerServices.Unsafe
    Assembly Version: 5.0.0.0
    Win32 Version: 5.0.20.51904
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Runtime.CompilerServices.Unsafe.DLL
----------------------------------------
System.Buffers
    Assembly Version: 4.0.3.0
    Win32 Version: 4.6.28619.01
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Buffers.DLL
----------------------------------------
System.Numerics.Vectors
    Assembly Version: 4.1.4.0
    Win32 Version: 4.6.26515.06
    CodeBase: file:///C:/Users/Anjan/Downloads/crc-tray(1)/System.Numerics.Vectors.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.



```

**Comment by gbraad** [comment]
@anjannath memory ... as it seems to be a set on the return of this

**Comment by anjannath** [comment]
@gbraad Ahh, yeah that is also an issue, but that only occurs if the daemon is not running, as we get the default minimum memory from the daemon, this exception was related to the structs getting changed during refactoring i think.

The daemon expects :`{"command":"setconfig", "args":{"properties":{"cpus":5"}}}` but the tray was sending `{"command":"setconfig", "args":{"cpus":5"}}`

**Comment by gbraad** [comment]
yep.. that was a VERY late modification 🤦

**Comment by anjannath** [comment]
Fixed in #62
